### PR TITLE
Conversions: Added more digital conversions and data transfer conversions

### PR DESCRIPTION
**Pull request by Lord-Lavios**
## Description of new Instant Answer, or changes
Introduces 1024 versions of digital conversions.

## Related Issues and Discussions
None

## People to notify
@pjhampton @moollaza 

## Testing & Review
To be completed by Community Leader (or DDG Staff) when reviewing Pull Request

**Pull Request**
- [ ] Title follows correct format (Specifies Instant Answer + Purpose)
- [ ] Description contains a valid Instant Answer Page Link (e.g. https://duck.co/ia/view/my_ia)

**Instant Answer Page** (for new Instant Answers)
- [ ] Instant Answer page is correctly filled out and contains:
    - The **Title** is appropriately named and formatted
    - The **IA topics** are present and appropriate
    - The **Description** is clear and coherent
    - **Source Name** exists if applicable
    - All **Example Queries** trigger on [Beta](https://beta.duckduckgo.com/)

**Code**
- [ ] Adheres to the [DuckDuckGo Style Guide](https://docs.duckduckhack.com/resources/code-style-guide.html)
- [ ] Behaviour is appropriately tested. If improvement, tests are adequately extended.
- [ ] There is no unnecessary files in place (such as editor config files)
- [ ] There is no API keys / secrets present
- [ ] Tests are passing (run `$ duckpan test <goodie_id>`)
    - Tester should report any failures

**Ready to merge?**

- [ ] Has this IA been deployed to and tested on [beta.duckduckgo.com](https://beta.duckduckgo.com/)?
- [ ] For larger commits, has this been approved be more than one community member?
- [ ] The number of reviews is appropriate for this type of PR
- [ ] The commit message is clear, coherent and fitting

**Pull Request Review Guidelines**: https://docs.duckduckhack.com/programming-mission/pr-review.html

---

Instant Answer Page: https://duck.co/ia/view/conversions

### CODE REVIEW

**Comment by daxtheduck** [comment]
Metadata placeholder post
            
*If you haven't already, please edit your first comment to include a link to the IA page at the bottom.*
            
---
            
**How to find your IA page**
- New Instant Answer
  -- follow the steps [here](https://duck.co/ia/new_ia) to create your IA page.
              
- Existing Instant Answer
  -- find your IA page from this [list](https://duck.co/ia)

**Comment by moollaza** [comment]
@Lord-Lavios we just made some changes to the units + testst, can you please update your code to resolve the conflicts?

File: share/goodie/conversions/conversions.js [hunk 1]
```diff
@@ -55,11 +55,23 @@ DDH.conversions = DDH.conversions || {};
...
             math.createUnit('gbit', '1000000000 b');
             math.createUnit('tbit', '1000000000000 b');
+            math.createUnit('pbit', '1000000000000000 b');
             math.createUnit('KB',   '1000 B');
             math.createUnit('MB',   '1000000 B');
             math.createUnit('GB',   '1000000000 B');
             math.createUnit('TB',   '1000 GB');
             math.createUnit('PB',   '1000 TB');
+            //1024 digital units
+            math.createUnit('kibit', '1024 b');
```
Comment (mintsoft) [inline share/goodie/conversions/conversions.js]: We should support Kib and Kibibit too

**Comment by Lord-Lavios** [comment]
@moollaza I've resolved the conflicts.

**Comment by mintsoft** [comment]
@Lord-Lavios It looks like you've got changes for multiple IA's in this. Can you rebase so the result of merging the PR would be to only update the Conversions IA?

File: share/goodie/conversions/conversions.js [hunk 2]
```diff
@@ -49,15 +49,35 @@ DDH.conversions = DDH.conversions || {};
...
+			{name: 'pibit', factor: '1125899906842624 b'},
+			{name: 'KiB',   factor: '1024 B'},
+			{name: 'MiB',   factor: '1048576 B'},
+			{name: 'GiB',   factor: '1073741824 B'},
+			{name: 'TiB',   factor: '1024 GiB'},
+			{name: 'PiB',   factor: '1024 TiB'},
+
+            //CUSTOM DATA RATE TRANSFER
+            {name: 'kbitps', factor: '1000 b'},
+			{name: 'mbitps', factor: '1000000 B'}, //Reversed mbitps to B because people type mbps and mean mbps
```
Comment (mintsoft) [inline share/goodie/conversions/conversions.js]: I would suggest caution making assumptions around people's intent without evidence, we have issues that have been solved previously in which users have defined their wishes. It was previously encapsulated within the back-end unit tests, however they're not applicable to this implementation so it's difficult to make that obvious, however we have the issues:
* https://github.com/duckduckgo/zeroclickinfo-goodies/issues/3990
* https://github.com/duckduckgo/zeroclickinfo-goodies/issues/3955#issuecomment-298236267

Comment (mintsoft) [inline share/goodie/conversions/conversions.js]: Also can you fix the indentation here?

Comment (Lord-Lavios) [inline share/goodie/conversions/conversions.js]: This is weird. Commit shows that the file is not indented correctly but when I view the file it is indented correctly, locally and on GitHub. Am I imagining things or is that really the case?

Comment (Lord-Lavios) [inline share/goodie/conversions/conversions.js]: I think we should do the same as we did with the data storage conversions. Both of the conversions assumes something from the user. Later, when the case-sensitive triggering is implemented both of the issues will be resolved.

Comment (pjhampton) [inline share/goodie/conversions/conversions.js]: @Lord-Lavios 

Try using a different text editor. This happens to me in vim.
As for the units, you guys can use your best judgment here.

Comment (mintsoft) [inline share/goodie/conversions/conversions.js]: > Later, when the case-sensitive triggering is implemented both of the issues will be resolved.
That's exactly the decision we made last time, which we've literally just fixed; it doesn't seem great to regress so soon, however it's up to you

**Comment by mintsoft** [comment]
@Lord-Lavios it appears you've converted spaces into tabs; we use spaces as our indentation at 4 spaces per tab

**Comment by Lord-Lavios** [comment]
@mintsoft 😅 Sorry about that. Can you confirm if it's okay now?

**Comment by pjhampton** [comment]
Closing as this has gone stale. It looks like these were added in another PR.
Thanks for your help as usual 👍

**Pull request closed without merging**